### DISCUSSION TITLE
Curl API integration in configuration file

### DISCUSSION THREAD

**Discussion by neerajasjawali** [edgexfoundry/community#114] [Q&A] [answered]
Hi All,

I am working with jakarta LTS. Earlier I was posting the subscription, secrets and notification using the curl commands. Now I am trying to post the subscription and secrets by adding these API in the support-notifications configuration.toml file. Below I have added the subscription in the .toml file.
```
[Subscription]
Name = "subscriptionA"  
 [Channels]
  Type = "EMAIL"
  Recipient = " neeraja@gmail.com"
Categories = "daily-notify"
Labels = "metadata"
Receiver = "receiverA"
ResendLimit = 0
ResendInterval = "2m"
AdminState = UNLOCKED"

```
Help me correctly define the subscription post API in support-notifications configuration.toml file.

Thanks in advance.

**Comment by cloudxxx8** [comment, marked as answer]
support-notifications doesn't support pre-defined Subscription, so this configuration section will be ignored.

**Reply by neerajasjawali** [reply]
> support-notifications doesn't support pre-defined Subscription, so this configuration section will be ignored.

@cloudxxx8 Okay. Then is it possible to pre-define the secrets?

**Reply by cloudxxx8** [reply]
Yes, you can refert to
https://github.com/edgexfoundry/app-service-configurable/blob/6891b07bca9ffb23389de7e5af914c7b678e0717/res/external-mqtt-trigger/configuration.toml#L38-L45C21

**Reply by neerajasjawali** [reply]
> Yes, you can refert to https://github.com/edgexfoundry/app-service-configurable/blob/6891b07bca9ffb23389de7e5af914c7b678e0717/res/external-mqtt-trigger/configuration.toml#L38-L45C21

@cloudxxx8, The link which you have provided is for app-service secrets. Can I define the secrets in the same way for support-notifications? The secrets I am referring to are the smtp secrets (email-id and app-password).

**Reply by cloudxxx8** [reply]
yes, you can try. if you define secret in the config, it is always insecure.

**Reply by neerajasjawali** [reply]
> yes, you can try. if you define secret in the config, it is always insecure.

I have tried defining the secrets as you suggested. But I couldn't get any notification. 

```
[Writable]
LogLevel = "INFO"
ResendLimit = 2
ResendInterval = "5s"
  [Writable.InsecureSecrets]
    [Writable.InsecureSecrets.DB]
    path = "redisdb"
      [Writable.InsecureSecrets.DB.Secrets]
      username = ""
      password = ""
    [Writable.InsecureSecrets.SMTP]
    path = "smtp"
      [Writable.InsecureSecrets.SMTP.Secrets]
      username = "neeraja@gmail.com"
      password = "vbibxznekyvwhdjg"
```

**Reply by neerajasjawali** [reply]
@cloudxxx8,I am able to get the notification only if I post secrets API manually.

**Reply by cloudxxx8** [reply]
my bad. it looks like support-notifications doesn't support this mechanism

**Reply by neerajasjawali** [reply]
Okay I understand. Thank you for your help @cloudxxx8
